src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -79,6 +84,11 @@ const onNumberOfInstancesChange = (
   } else {
     currentValues.set("warningBanner", undefined);
   }
+
+  currentValues.set("costPerHour", {
```

Comment:
set costPerHour in the onEnableDedicatedGatewayChange function as well, based on whether 'newValue' is true or false. calculate in the true case, hide in false case.